### PR TITLE
Added Officer's Sabre suicide

### PR DESCRIPTION
**Pull request by cinder1992**
:cl: cinder1992
add: The Captain can now go down with their station in style! Suicide animation added to the Officer's Sabre.
/:cl:

Honestly, I found it quite disappointing that those captains (or assistants) wanting to go max condom just got the usual suicide spiel whenever they suicided with their sabre. This corrects that oversight by adding an animation that dismembers the user's limbs before killing them outright. honestly there's less blood than I'd like there to be, but I feel it's more than good enough as-is.

### CODE REVIEW

**Comment by Bawhoppen** [comment]
I honestly thought you were Jordie based on your avatar before I saw the username.

**Comment by cinder1992** [comment]
I haven't used this account in forever, I changed the avatar to be consistent with the forums/discord but I guess it hasn't reached all of github's CDN caches.

File: code/game/objects/items/melee/misc.dm [hunk 1]
```diff
@@ -88,6 +88,28 @@
 	if(istype(B))
 		playsound(B, 'sound/items/sheath.ogg', 25, 1)
 
+/obj/item/melee/sabre/suicide_act(mob/user)
+	user.visible_message("<span class='suicide'>[user] is trying to cut off all [user.p_their()] limbs with [src]! it looks like [user.p_theyre()] trying to commit suicide!</span>")
+	var/list/bodyparts = list(BODY_ZONE_L_ARM, BODY_ZONE_R_ARM, BODY_ZONE_L_LEG, BODY_ZONE_R_LEG)
+	for(var/i = 1 to 4)
+		var/picked = pick(bodyparts)
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/objects/items/melee/misc.dm]: pick_n_take?

Comment (ExcessiveUseOfCobblestone) [reply code/game/objects/items/melee/misc.dm]: yeah use the macro instead.

Comment (cinder1992) [reply code/game/objects/items/melee/misc.dm]: was not aware that was a thing, now I am aware it is a thing, and I am making it a thing in the pr code.

File: code/game/objects/items/melee/misc.dm [hunk 2]
```diff
@@ -88,6 +88,26 @@
...
+	var/obj/item/bodypart/affecting = user.get_bodypart(zone)
+	if(affecting && affecting.dismemberable)
+		affecting.dismember(BRUTE)
+		user.adjustBruteLoss(20)
+	playsound(user, 'sound/weapons/rapierhit.ogg', 25, 1)
+
+/obj/item/melee/sabre/proc/manual_suicide(mob/living/user)
+	playsound(user, 'sound/weapons/rapierhit.ogg', 25, 1)
+	user.adjustBruteLoss(200)
+	user.death(0)
```
Comment (Cyberboss) [inline code/game/objects/items/melee/misc.dm]: 0 -> FALSE

File: code/game/objects/items/melee/misc.dm [hunk 3]
```diff
@@ -88,6 +88,26 @@
 	if(istype(B))
 		playsound(B, 'sound/items/sheath.ogg', 25, 1)
 
+/obj/item/melee/sabre/suicide_act(mob/user)
+	user.visible_message("<span class='suicide'>[user] is trying to cut off all [user.p_their()] limbs with [src]! it looks like [user.p_theyre()] trying to commit suicide!</span>")
+	var/list/bodyparts = list(BODY_ZONE_L_ARM, BODY_ZONE_R_ARM, BODY_ZONE_L_LEG, BODY_ZONE_R_LEG)
+	for(var/i = 1 to 4)
```
Comment (Cyberboss) [inline code/game/objects/items/melee/misc.dm]: for(var/i in 1 to 4)

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by DaxDupont** [comment]
This assumes there's only 4 limbs.

**Comment by cinder1992** [comment]
This is true, however, there's no sane way that I can see that allows you to dismember an arbitrary limb without giving it a known target. It handles fewer than four limbs by making sure the limb exists, but handling more than four would require some... interesting coding.

File: code/game/objects/items/melee/misc.dm [hunk 4]
```diff
@@ -88,6 +88,26 @@
...
+	var/list/bodyparts = list(BODY_ZONE_L_ARM, BODY_ZONE_R_ARM, BODY_ZONE_L_LEG, BODY_ZONE_R_LEG)
+	for(var/i in 1 to 4)
+		addtimer(CALLBACK(src, .proc/suicide_dismember, user, pick_n_take(bodyparts)), 10 * i)
+	addtimer(CALLBACK(src, .proc/manual_suicide, user), 50)
+	return MANUAL_SUICIDE
+
+/obj/item/melee/sabre/proc/suicide_dismember(mob/living/user, zone)
+	var/obj/item/bodypart/affecting = user.get_bodypart(zone)
+	if(affecting && affecting.dismemberable)
+		playsound(user, 'sound/weapons/rapierhit.ogg', 25, 1)
```
Comment (Cyberboss) [inline code/game/objects/items/melee/misc.dm]: change res to `hitsound`

File: code/game/objects/items/melee/misc.dm [hunk 5]
```diff
@@ -88,6 +88,26 @@
...
+		affecting.dismember(BRUTE)
+		user.adjustBruteLoss(20)
+
+/obj/item/melee/sabre/proc/manual_suicide(mob/living/user)
+	playsound(user, 'sound/weapons/rapierhit.ogg', 25, 1)
```
Comment (Cyberboss) [inline code/game/objects/items/melee/misc.dm]: ditto

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by JJRcop** [comment]
The arm that holds the sabre should be cut last

File: code/game/objects/items/melee/misc.dm [hunk 6]
```diff
@@ -88,6 +88,26 @@
...
+
+/obj/item/melee/sabre/proc/suicide_dismember(mob/living/user, zone)
+	var/obj/item/bodypart/affecting = user.get_bodypart(zone)
+	if(affecting && affecting.dismemberable)
+		playsound(user, hitsound, 25, 1)
+		affecting.dismember(BRUTE)
+		user.adjustBruteLoss(20)
+
+/obj/item/melee/sabre/proc/manual_suicide(mob/living/user)
+	playsound(user, hitsound, 25, 1)
```
Comment (JJRcop) [inline code/game/objects/items/melee/misc.dm]: Should add ```if(!QDELETED(user))```

**Comment by demonfiren** [comment]
it's not the corporate seppuku I expected but okay

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**